src/swap/sideshift.js

```diff
@@ -0,0 +1,310 @@
+      }
+      if (CURRENCY_CODE_TRANSCRIPTION[request.toCurrencyCode]) {
+        safeToCurrencyCode = CURRENCY_CODE_TRANSCRIPTION[request.toCurrencyCode]
+      }
+
+      const rate: Rate = await get(
+        `pairs/${safeFromCurrencyCode}/${safeToCurrencyCode}`
+      )
+
+      const quoteAmount =
+        request.quoteFor === 'from'
+          ? await request.fromWallet.nativeToDenomination(
+              request.nativeAmount,
+              request.fromCurrencyCode
+            )
+          : await request.toWallet.nativeToDenomination(
+              request.nativeAmount,
+              request.toCurrencyCode
+            )
+
+      const depositAmount =
+        request.quoteFor === 'from'
+          ? quoteAmount
+          : (parseFloat(quoteAmount) / rate.rate).toFixed(8).toString()
+
+      await checkRateForError(rate, request, depositAmount)
+
+      const fixedRateQuoteParams: FixedQuoteRequestParams = {
+        depositMethod: safeFromCurrencyCode,
+        settleMethod: safeToCurrencyCode,
+        depositAmount
+      }
+
+      const fixedRateQuote: FixedQuote = await post(
+        'quotes',
+        fixedRateQuoteParams
+      )
+
+      const orderRequestParams: OrderRequestParams = {
+        type: 'fixed',
+        quoteId: fixedRateQuote.id,
+        affiliateId: initOptions.affiliateId,
+        settleAddress
+      }
+
+      const quoteInfo: OrderRequest = await post('orders', orderRequestParams)
+
+      const spendInfoAmount = await request.fromWallet.denominationToNative(
+        quoteInfo.depositAmount,
+        request.fromCurrencyCode.toUpperCase()
+      )
+
+      const amountExpectedFromNative = await request.fromWallet.denominationToNative(
+        quoteInfo.depositAmount,
+        request.fromCurrencyCode
+      )
+
+      const amountExpectedToNative = await request.fromWallet.denominationToNative(
+        quoteInfo.settleAmount,
+        request.toCurrencyCode
+      )
+
+      const spendInfo: EdgeSpendInfo = {
+        currencyCode: request.fromCurrencyCode,
+        spendTargets: [
+          {
+            nativeAmount: spendInfoAmount,
+            publicAddress: quoteInfo.depositAddress.address
+          }
+        ],
+        networkFeeOption:
+          request.fromCurrencyCode.toUpperCase() === 'BTC'
+            ? 'high'
+            : 'standard',
+        swapData: {
+          orderId: quoteInfo.orderId,
+          isEstimate: false,
+          payoutAddress: settleAddress,
+          payoutCurrencyCode: safeToCurrencyCode,
+          payoutNativeAmount: amountExpectedToNative,
+          payoutWalletId: request.toWallet.id,
+          plugin: { ...swapInfo },
+          refundAddress: depositAddress
+        }
+      }
+      const tx: EdgeTransaction = await request.fromWallet.makeSpend(spendInfo)
+
+      const orderExpirationMs = 1000 * 60 * 5
+      const orderExpirationDate =
+        new Date(quoteInfo.expiresAtISO).getTime() + orderExpirationMs
```

Comment:
Yikes, are we returning `createdAtISO` as `expiresAtISO` from the API?

Reply:
Found the bug on SideShift.ai side. Fix will be live today or early tomorrow. @szgergely @nilswx